tests/ts_cluster_updowngrade_select_aggregation_SUITE.erl

```diff
@@ -0,0 +1,243 @@
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(ts_cluster_updowngrade_select_aggregation_SUITE).
+
+-export([
+         suite/0,
+         init_per_suite/1,
+         end_per_suite/1,
+         all/0,
+         run_this_test/1
+        ]).
+
+-ifdef(EQC).
+-include_lib("eqc/include/eqc.hrl").
+-compile(export_all).
+-endif.
+
+-include_lib("common_test/include/ct.hrl").
+-include("ts_updown_util.hrl").
+
+-define(TABLE, "updown_aggregation_test_table").
+-define(TEMPERATURE_COL_INDEX, 4).
+-define(PRESSURE_COL_INDEX, 5).
+-define(PRECIPITATION_COL_INDEX, 6).
+
+%% Callbacks
+
+suite() ->
+    [{timetrap, {seconds, 9000}}].
+
+all() ->
+    [
+     run_this_test
+    ].
+
+init_per_suite(Config) ->
+    lists:foldl(
+      fun(Fun, Cfg) -> Fun(Cfg) end,
+      Config,
+      [fun ts_updown_util:setup/1,
+       fun make_scenario_invariants/1]).
+
+end_per_suite(Config) ->
+    ts_updown_util:maybe_shutdown_client_node(Config).
+
+
+-ifdef(EQC).
+
+%% it's 2 to the power of the number of binary scenario parameters in
+%% #scenario{} (6)
+-define(NUMCASES, 64).
+
+run_this_test(Config) ->
+    case eqc:quickcheck(
+           eqc:numtests(?NUMCASES, prop_scenario(Config))) of
+        true ->
+            pass;
+        _ ->
+            ct:fail("There were failing queries", [])
+    end.
+
+prop_scenario(Config) ->
+    ?FORALL(
+       Sce, gen_scenario(),
+       [] == ts_updown_util:run_scenario(Config, Sce)).
+
+gen_scenario() ->
+    ?LET({TableNodeVsn, QueryNodeVsn,
+          NeedTableNodeTransition, NeedQueryNodeTransition,
+          NeedPreClusterMixed, NeedPostClusterMixed,
+          Table},
+         {gen_version(), gen_version(),
+          bool(), bool(), bool(), bool(),
+          gen_table()},
+         #scenario{table_node_vsn = TableNodeVsn,
+                   query_node_vsn = QueryNodeVsn,
+                   need_table_node_transition = NeedTableNodeTransition,
+                   need_query_node_transition = NeedQueryNodeTransition,
+                   need_pre_cluster_mixed = NeedPreClusterMixed,
+                   need_post_cluster_mixed = NeedPostClusterMixed,
+                   table = Table,
+                   ddl = ts_util:get_ddl(aggregation, binary_to_list(Table))}).
+
+gen_version() ->
+    ?LET(A, oneof([current, previous]), A).
+
+gen_table() ->
```

Comment:
You will get random crashes if you use gen_name to generate a table name because gen_name doesn't return a unique set - the answer is to not specify the table name in the scenario record at all (nor substitute it into the CREATE TABLE, INSERT INTO or SELECT queries until run-time - and then use a timestamp-to-string...